### ISSUE TITLE
bug(oracle): MoE experts fail to train due to matrix singularity - insufficient samples

### ISSUE THREAD

**Issue by noahgift** [paiml/depyler#154]
## Summary

All 4 MoE Oracle experts fail to train and fall back to hardcoded default patterns due to insufficient training samples.

## Symptoms

When running oracle tests:
```
Warning: Expert SyntaxBorrowing training failed (matrix singularity), using default patterns: Insufficient samples
Warning: Expert ScopeResolution training failed (matrix singularity), using default patterns: Matrix is not positive definite
Warning: Expert MethodField training failed (matrix singularity), using default patterns: Matrix is not positive definite
Warning: Expert TypeSystem training failed (matrix singularity), using default patterns: Matrix is not positive definite
```

## Impact

- **Tests pass** but only because hardcoded defaults handle test cases
- **No actual ML learning** is happening
- **Accuracy is unmeasured** - oracle is essentially rule-based, not ML-based
- 98% of training data is synthetic variations, real data is ~212 samples

## Root Cause

`LinearRegression` in each expert requires `n_samples >= n_features + 1`. Current sample counts per expert domain are below this threshold.

### Short-term (workaround)
1. Switch from `LinearRegression` to `Ridge` regression (handles singularity via regularization)
2. Or reduce feature dimensionality

### Long-term (proper fix)
1. Close #153 to add real training data from reprorusted-python-cli
2. This will provide 3000+ real samples, well above the threshold

## Location

- `crates/depyler-oracle/src/moe_oracle.rs` - expert training
- `crates/depyler-oracle/src/unified_training.rs` - data pipeline

## Acceptance Criteria

- [ ] No matrix singularity warnings during training
- [ ] Experts actually learn from data (not just defaults)
- [ ] Add accuracy metric logging after training
- [ ] Accuracy >= 85% on held-out test set

## Related

- #153 - Automated training pipeline (will provide more data)
- aprender `Ridge` regression available in `aprender::linear_model::Ridge`
